web/src/main/java/org/wso2/testgrid/web/sso/TestgridSSOAgentFilter.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.testgrid.web.sso;
+
+import org.apache.commons.lang.StringUtils;
+import org.glassfish.jersey.internal.util.Base64;
+import org.wso2.carbon.identity.sso.agent.SSOAgentConstants;
+import org.wso2.carbon.identity.sso.agent.SSOAgentFilter;
+import org.wso2.carbon.identity.sso.agent.bean.SSOAgentConfig;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+
+/**
+ * Implementation of the SSOAgent Filter which identifies the correct httpBinding
+ * to send the request to Identity Provider.
+ */
+public class TestgridSSOAgentFilter extends SSOAgentFilter {
+
+    private static Logger logger = Logger.getLogger("org.wso2.testgrid.web.sso.TestgridSSOAgentFilter");
+
+    private static final String USERNAME = "username";
+    private static final String PASSWORD = "password";
```

Comment:
I'm not sure whether these default values r correct. Why do you need a password in this filter anyway?